### PR TITLE
adding ci-knative-serving-contour-latest to beta-prow-tests

### PR DESCRIPTION
**Pull request by peterfeifanchen**
we want to test Go compiler versions without breaking prod

/cc @chizhg @mattmoor

### CODE REVIEW

File: config/prod/prow/jobs/custom/experimental.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - image: gcr.io/knative-tests/test-infra/prow-tests:beta
+      imagePullPolicy: Always
+      command:
+      - runner.sh
+      args:
+      - "./test/presubmit-tests.sh"
+      - "--run-test"
+      - "./test/e2e-tests.sh --contour-version latest"
+      - "--run-test"
+      - "./test/e2e-auto-tls-tests.sh --contour-version latest"
```
Comment (mattmoor) [inline config/prod/prow/jobs/custom/experimental.yaml:40]: We don't need these

**Review by mattmoor — COMMENTED** [review]
/lgtm
/hold

In case you want to remove the excess (it provisions another cluster, and isn't needed to repro the bug)

File: config/prod/prow/jobs/custom/experimental.yaml (hunk 1 above)
Comment (peterfeifanchen) [reply config/prod/prow/jobs/custom/experimental.yaml:40]: any of the args or just the second set of --run-test? I just copied the original contour ci.

**Review by mattmoor — COMMENTED** [review]
/lgtm

**Pull request merged**